archunit/src/main/java/com/tngtech/archunit/lang/ArchRule.java

```diff
@@ -215,6 +222,16 @@ public EvaluationResult evaluate(JavaClasses classes) {
                 return new EvaluationResult(this, events, priority);
             }
 
+            private void verifyNotEmptyIfActive(Iterable<T> allObjects) {
+                if (isPropertyFailOnEmptyShouldActive() && Iterables.isEmpty(allObjects)) {
+                    throw new AssertionError("Rule failed to check any classes");
```

Comment:
As mentioned, I was thinking about sth. like "Rule failed to check any classes. To allow rules being executed without checking any classes adjust the ArchUnit configuration to `archRule.failOnEmptyShould=false`" or sth. like that.